### PR TITLE
Feature/edit address

### PR DESCRIPTION
**Pull request by moroz0v**

### CODE REVIEW

File: app/src/main/java/nyc/ignitelabs/civicduty/IntentUtility.kt [hunk 1]
```diff
@@ -0,0 +1,13 @@
+package nyc.ignitelabs.civicduty
+
+import android.content.Context
+import android.content.Intent
+import nyc.ignitelabs.civicduty.view.EditActivity
+
+fun  addressEditIntent(context: Context, address: String ) : Intent {
```
Comment (phyllisstein) [inline app/src/main/java/nyc/ignitelabs/civicduty/IntentUtility.kt:7]: What's your thinking behind `IntentUtility` + anonymous functions? I can see three alternatives, off the top of my head:

1. Intents are built up during construction of the class meant to fire them, and assigned to properties.
2. Intents are built up lazily by class methods in the class meant to fire them.
3. Intents are generated by anonymous methods like these, but with narrower responsibilities---say, `nyc.ignitelabs.civicduty.GreetingIntentUtility`.

I feel like any one of the four (or more) possibilities can be justified, so it'd just be fun to get at your reasoning.

File: app/src/main/java/nyc/ignitelabs/civicduty/Constants.kt [hunk 2]
```diff
@@ -0,0 +1,6 @@
+package nyc.ignitelabs.civicduty
+
+object Constants{
```
Comment (phyllisstein) [inline app/src/main/java/nyc/ignitelabs/civicduty/Constants.kt:3]: 😗👌

File: app/src/main/java/nyc/ignitelabs/civicduty/view/EditActivity.kt [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+
+class EditActivity : AppCompatActivity() {
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+
+        setContentView(R.layout.activity_edit)
+
+        button_submit.setOnClickListener {
+            setResult(RESULT_OK, addressEditSuccessIntent(address_edit.text.toString()))
```
Comment (phyllisstein) [inline app/src/main/java/nyc/ignitelabs/civicduty/view/EditActivity.kt:20]: What is the type of `address_edit.text` before `#toString`?

File: app/src/main/java/nyc/ignitelabs/civicduty/view/MainActivity.kt [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+import nyc.ignitelabs.civicduty.addressEditIntent
+
+class MainActivity : AppCompatActivity() {
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+
+        setContentView(R.layout.activity_main)
+
+        button_submit.setOnClickListener {
```
Comment (phyllisstein) [inline app/src/main/java/nyc/ignitelabs/civicduty/view/MainActivity.kt:20]: This is so much cleaner than the `findViewById` path; is it considered idiomatic?

File: app/src/main/java/nyc/ignitelabs/civicduty/IntentUtility.kt (hunk 1 above)
Comment (moroz0v) [reply app/src/main/java/nyc/ignitelabs/civicduty/IntentUtility.kt:7]: Idea was to move this code outside of the Activity class for now. As we apply arch patterns, I think it'll end up being close to your 3rd point! Good eye man

File: app/src/main/java/nyc/ignitelabs/civicduty/view/EditActivity.kt (hunk 3 above)
Comment (moroz0v) [reply app/src/main/java/nyc/ignitelabs/civicduty/view/EditActivity.kt:20]: `Editable` interface, which is just an instance of a `SpannableString`.

File: app/src/main/java/nyc/ignitelabs/civicduty/view/MainActivity.kt (hunk 4 above)
Comment (moroz0v) [reply app/src/main/java/nyc/ignitelabs/civicduty/view/MainActivity.kt:20]: So this is possible only with Kotlin extensions for Android. This is the line that enables it: `import kotlinx.android.synthetic.main.activity_edit.*`

There is a library that's used explicitly for this, which is called `ButterKnife` and which LUNA project uses but it's essentially enabling this very similar functionality for our purposes so I just skipped it in favor of this. It's super neat and ability to do s**t like this is why Android devs get excited about using Kotlin!

File: app/src/main/java/nyc/ignitelabs/civicduty/IntentUtility.kt (hunk 1 above)
Comment (phyllisstein) [reply app/src/main/java/nyc/ignitelabs/civicduty/IntentUtility.kt:7]: Totally makes sense. Thanks!

**Pull request merged**